include/cetl/pmr/interface_ptr.hpp

```diff
@@ -0,0 +1,90 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PMR_INTERFACE_PTR_H_INCLUDED
+#define CETL_PMR_INTERFACE_PTR_H_INCLUDED
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+#include <memory>
+#include <functional>
+
+namespace cetl
+{
+namespace pmr
+{
+
+template <typename Interface>
+class PmrInterfaceDeleter final
+{
+public:
+    template <typename PmrAllocator>
+    PmrInterfaceDeleter(PmrAllocator alloc, std::size_t obj_count)
```

Comment:
Why are you passing the allocator by value here and also capturing it by value?

Reply:
As far as I see (at std, and llvm & gcc implementation) it's common practice to pass allocators and deleters by value. I think this is done to eliminate "ownership" and lifetime issues.

Reply:
Okay; I thought about pmr semantics here but it doesn't really apply. It is not appropriate to pass a pmr by value since it's polymorphic, but this is just an allocator. This raises another question: does it really need to be a PMR allocator? The code is generic and should be equally applicable to an allocator that does not use PMR, so do we need `Pmr` in the name?

Reply:
"Pmr" here b/c in reality it will be `cetl::pmr::polymorphic_allocator<Concrete>`, and its `deallocate` method will be in use on interface ptr ~destructor (or `reset`). Also, allocator itself is not polymorphic - internally used `cetl::pmr::memory_resourse` is.